### PR TITLE
Challengers now receive the identity of their target a minute after roundstart

### PR DESCRIPTION
**Pull request by DeityLink**
:cl:
* tweak: Challengers now receive the identity of their target a minute after roundstart.

### CODE REVIEW

**Comment by Kurfursten** [comment]
We already have a delayed reveal kill objective, why not use that or make a subtype??

**Comment by DeityLink** [comment]
> 
> 
> We already have a delayed reveal kill objective, why not use that or make a subtype??

Didn't know that type existed but I'm not sure if I want to redo all the adjustments I made to /datum/objective/target/assassinate to another type for the same result I get here with 2 lines, and without the risk of messing up other features.

**Comment by Kurfursten** [comment]
Avoiding two types that do the same thing makes the code more maintainable, and presumably it should be a fairly similar linecount. What tweaks would you want to make to one assassinate that would not be better made available to all versions of assassinate or objectives more broadly?

**Comment by DeityLink** [comment]
> Avoiding two types that do the same thing makes the code more maintainable, and presumably it should be a fairly similar linecount. What tweaks would you want to make to one assassinate that would not be better made available to all versions of assassinate or objectives more broadly?

Now that I think about it'd be best indeed to have only one assassinate objective type that can have a delay, set to zero by default. I'll do that tomorrow.

**Comment by DeityLink** [comment]
Tomorrow comes today. All target objectives can now have a delay.

**Comment by Kurfursten** [comment]
Check compiler

**Comment by DeityLink** [comment]
right

File: code/datums/gamemode/objectives/target/assassinate.dm [hunk 1]
```diff
@@ -4,6 +4,14 @@ var/list/assassination_objectives = list()
 	name = "Assassinate <target>"
 	var/syndicate_checked = 0
 
+/datum/objective/target/assassinate/delay_short
+	name = "Assassinate <target> after 1 minute"
+	delay = 60 SECONDS
+
+/datum/objective/target/assassinate/delay_medium
+	name = "Assassinate <target> after 10 minute"
```
Comment (Kurfursten) [inline code/datums/gamemode/objectives/target/assassinate.dm]: s

File: code/datums/gamemode/objectives/target/target.dm [hunk 2]
```diff
@@ -22,20 +15,31 @@
...
+	owner.current.playsound_local(owner.current.loc, 'sound/machines/twobeep.ogg', 75, 0)
+	to_chat(owner.current, "<span class='userdanger'>Target Revealed</span>: <b>[explanation_text]</b>")
+
 /datum/objective/target/proc/find_target()
 	var/list/possible_targets = get_targets()
 	if(possible_targets.len > 0)
-		target = pick(possible_targets)
+		if (delay)
+			delayed_target = pick(possible_targets)
+			to_chat(owner.current, "<span class='danger'>Your target's identity will be revealed to you in [delay/600] MINUTE[(delay > 600) ? "S":""].</span>")
```
Comment (Kurfursten) [inline code/datums/gamemode/objectives/target/target.dm:40]: No round? Regardless we may have a helper to turn this into minutes and seconds remaining.

Comment (Kurfursten) [reply code/datums/gamemode/objectives/target/target.dm:40]: Nevermind, this isn't decrementing, duh

File: code/datums/gamemode/objectives/target/assassinate.dm (hunk 1 above)
Comment (DeityLink) [reply code/datums/gamemode/objectives/target/assassinate.dm]:
```suggestion
	name = "Assassinate <target> after 10 minutes"
```

**Comment by DeityLink** [comment]
s

**Pull request merged**